MetaMask/metamask-extension
# Improvements for Smart Transactions / Swaps

## Explanation
This PR includes: 
- Don’t call /estimateGas if a user doesn’t have enough funds
- Hardcode block explorer URLs for Swaps
- Track the "stx_prev_user_opt_in" param
- Add fee estimates tracking for regular txs and STX
- Track estimated_gas and estimated_vs_used_gasRatio for STX
- Only track the "Error Smart Transactions" event once
- Don't overwrite "maxGasLimit" for STX on the View Quote page for better "balance needed" estimations
- Update description for Transak
- Fix styles for the input field on the Build Quote page
- Refactor variables for STX error types and add translation for each STX error type
- Do additional logging for the "current_stx_enabled" param
- Add a close icon for an STX notification, update STX content

## Manual testing steps
Don’t call /estimateGas if a user doesn’t have enough funds:
- Enable STX
- On the Build Quote page select 0 in the input field, select tokens and submit the form
- On the View Quote page you can check the Network tab. It doesn't make a call for the /estimateGas API

Hardcode block explorer URLs for Swaps:
- Select `BNB` as the `Swap to` token on the Build Quote page. It will show you a notification `Only verified on 1 source.`. Once you click on a link there, it will open a token detail page

Tracking:
- For tracking updates you can open the Network tab in the background dev tools and look for `/batch` API calls and their payloads
